OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -34,7 +34,7 @@ public Attack(Actor self, Target target, bool allowMovement, bool forceAttack)
 
 			this.forceAttack = forceAttack;
 
-			attack = self.Trait<AttackBase>();
+			attack = self.TraitsImplementing<AttackBase>().First(x => !x.IsTraitDisabled);
```

Comment:
What if that specific AttackBase becomes disabled between the constructor and tick (which may be quite far apart if the activity is queued)?